kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncProcessor.java

```diff
@@ -505,6 +544,8 @@ private int finalizeAtLeastOneEvent()  {
    */
   private void maybeBackOffEnqueuingNewEventWithKey(final KIn key)  {
     while (schedulingQueue.keyQueueIsFull(key)) {
+      // TODO: switch to a periodic log
+      log.debug("key queue is full. back off until there is room on key queue");
```

Comment:
nit: let's log which key is backing off, size of queue and back off time (if that's easily accessible)